lib/court_data_adaptor/resource/queryable.rb

```diff
@@ -19,6 +19,13 @@ module InstanceMethods
         def resource
           self.class.resource
         end
+
+        def refresh_token_if_required!
+          resource.connection(true) do |conn|
```

Comment:
Does this get called regardless of the token being active?

Reply:
Not sure exactly what magic is going on in either Faraday::Oauth2 or JsonApiClient::Connection but, from my code, it only gets a new token (calls 'oauth/token') if current token has `expired?`. 

From my manual local testing (with a 1 minute expiry time in the adaptor locally 😁 ) it only requests a new token on submission of a query if the previous has expired, even though this code always runs.  I am not sure of the impact of this code running on every query but network traffic wise it does not add any more round trips (except one extra when a new token is required, every 2 hours)

Reply:
fair enough, thanks for clarifying